packages/nextjs/src/middleware/withMiddlewareAuth.ts

```diff
@@ -1,15 +1,23 @@
 import { NextResponse } from 'next/server';
 import { NextMiddleware } from 'next/server';
-import { 
-  CookieOptions, 
-  setCookies, 
+import {
+  CookieOptions,
+  setCookies,
   COOKIE_OPTIONS,
-  TOKEN_REFRESH_MARGIN, 
+  TOKEN_REFRESH_MARGIN,
   NextRequestMiddlewareAdapter,
   NextResponseMiddlewareAdapter,
-  jwtDecoder
+  jwtDecoder,
+  User
 } from '@supabase/auth-helpers-shared';
 
+class NoPermissionError extends Error {
+  constructor(message: string) {
+    super(message);
+    Object.setPrototypeOf(this, NoPermissionError.prototype);
```

Comment:
Is this necessary? it seems to be advised against in the [MDN](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/setPrototypeOf) docs.

Reply:
ya, the typesafe and more readable variant would be with a simple override of the "name" getter. check it out in [this](https://www.typescriptlang.org/play?#code/MYGwhgzhAEByD2AFApgJwLYEsqfgOwFFVV5VpkAPAF2TwBMYiSyBvaAKGmmHwitQCuwKqQAU6ZFDABzZAC5ofVJjzSAlNBacuigQAc04yRBnI12gL7bZVaHjATRGtqmRUBqPNCoALbADoePCUhEVR-ewloKyt2UEgYAHlfNCZScmpaBmg01g4uIJDhMQkpWQUlFXVNbS4IfUNSk1lzLisuGzsHZCdNaFd3T28-CEDeflDSCO7o9lj+AE8argB6Fd8SAHc7ZG3c0QByMDx8BfR4ARg0ZgPW4a2d7YQUDGwIXEJiMQOT6ANXnD4chfVC3bQbeDbPC7aDJHypEGHeApMjXUhgizcMBUYA+aCia7ObSYABm+Ou0BUfGOwGQ8DJzzQWEBn2YRJ03F48BAyH8IHg0lEACIKfTvAsDNAhdAANTA8KRMyWcggCDISlkgnESnBKg0ulkuEItnLHSFbm8-mCkXasVUCXq6Vy67TCR3THIVXqrQc808vkC4UCPAAaxOmy8aLIdodUtl5O1kByILUrqVbTmQA) playground.

Reply:
I'm getting this warning though. 

> Accessors are only available when targeting ECMAScript 5 and higher